src/control/server/mgmt_system.go

```diff
@@ -408,3 +416,43 @@ func (svc *mgmtSvc) StartRanks(ctx context.Context, req *mgmtpb.RanksReq) (*mgmt
 
 	return resp, nil
 }
+
+// ClusterEvent management service gRPC handler receives ClusterEvent requests
+// from control-plane instances attempting to notify the MS of a cluster event
+// in the DAOS system.
+//
+// On receipt of the cluster event request, process any encapsulated RAS event
+// messages (handling of other types to be supported in the future).
+//
+// Necessary state updates for the rank involved in the event will be
+// registered in the system membership.
+func (svc *mgmtSvc) ClusterEvent(ctx context.Context, req *mgmtpb.ClusterEventReq) (*mgmtpb.ClusterEventResp, error) {
+	if req == nil {
+		return nil, errors.New("nil request")
+	}
```

Comment:
You could use `svc.checkLeaderRequest(req)` here to do the nil and leader checks in one go.